src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -227,20 +232,34 @@ class Queue[T <: Data](val gen: T,
   val do_deq = WireDefault(io.deq.fire())
 
   when (do_enq) {
-    ram(enq_ptr.value) := io.enq.bits
+    if (useSyncReadMem) {
+      ram.write(enq_ptr.value, io.enq.bits)
+    }
+    else {
+      ram(enq_ptr.value) := io.enq.bits
+    }
     enq_ptr.inc()
   }
   when (do_deq) {
     deq_ptr.inc()
+    if (useSyncReadMem)
+      deq_ptr_value.get := Mux(deq_ptr.value === (entries.U - 1.U), 0.U, deq_ptr.value + 1.U)
   }
+  
   when (do_enq =/= do_deq) {
     maybe_full := do_enq
   }
 
   io.deq.valid := !empty
   io.enq.ready := !full
-  io.deq.bits := ram(deq_ptr.value)
-
+  
+  if (useSyncReadMem) {
+    io.deq.bits := ram.read(deq_ptr_value.get)
```

Comment:
```suggestion
    val deq_addr = WireDefault(deq_ptr_value.get)
    io.deq.bits := ram.read(deq_addr)
```

An unconditionally connected wire will make the read unconditional as an easy, non-optimal-but-correct alternative.

Reply:
Thanks for pointing this out and suggesting the fix. It's included in the current commit.

Reply:
Thanks, I see that it correctly uses an unconditional read. Sorry for the confusion, it seems like I've been looking at multiple versions incorrectly, and I got a little confused with the new use of option + WireDefault at the top level.

I wrote a new non-line-comment to be a little easier to follow.

Reply:
Another great advice to limit the scope! Your idea is incorporated in the latest commit. Thank you.